src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/OffenderService.kt

```diff
@@ -50,6 +82,27 @@ OffenderService(
     }
   }
 
+  fun isXWorkingDaysBefore(days: Int, releaseDate: LocalDate?): Boolean {
+    if (releaseDate === null) return false
+    val dateBeforeXWorkingDays = getEarliestReleaseDate(days, releaseDate)
+    return LocalDate.now(clock).isEqual(dateBeforeXWorkingDays) || LocalDate.now(clock).isAfter(dateBeforeXWorkingDays)
+  }
+
+  private fun isEligibleForEarlyRelease(releaseDate: LocalDate?): Boolean {
+    val dayOfWeek = releaseDate?.dayOfWeek
+    if (dayOfWeek == DayOfWeek.SATURDAY || dayOfWeek == DayOfWeek.SUNDAY) {
+      return true
+    }
+    val listOfBankHolidays: List<LocalDate> = bankHolidayService.getBankHolidaysForEnglandAndWales()
+    return listOfBankHolidays.contains(releaseDate)
+  }
+
+  private fun getEarliestReleaseDate(days: Int, releaseDate: LocalDate) =
```

Comment:
Rather than duplicating all this, it should really sit in the releaseDateService.

Reply:
No the logic there is different so I added some flag before, but you suggested to create a new method instead.